gpioioctl/lineset.go

```diff
@@ -248,21 +256,28 @@ func (ls *LineSet) WaitForEdge(timeout time.Duration) (number uint32, edge gpio.
 	} else if event.Id == _GPIO_V2_LINE_EVENT_FALLING_EDGE {
 		edge = gpio.FallingEdge
 	}
-	number = uint32(event.Offset)
+	number = int(event.Offset)
 	return
 }
 
-// ByOffset returns a line by it's offset in the LineSet.
-func (ls *LineSet) ByOffset(offset int) *LineSetLine {
+// ByOffset returns a line by it's offset in the LineSet.  See ByName() for an
+// example that casts the return value to a LineSetLine
+func (ls *LineSet) ByOffset(offset int) pin.Pin {
 	if offset < 0 || offset >= len(ls.lines) {
 		return nil
 	}
 	return ls.lines[offset]
 }
 
-// ByName returns a Line by name from the LineSet.
-func (ls *LineSet) ByName(name string) *LineSetLine {
-
+// ByName returns a Line by name from the LineSet. To cast the returned value
+// to a LineSet line, use:
```

Comment:
That would be better as an example.

Reply:
Just to confirm, you're recommending duplicating the example on ByName() to the doc for ByOffset()?

Reply:
I was thinking about creating a toy example in example_test.go but it was really just a nit, don't bother.